force-app/main/default/classes/CON_DeleteContactOverride_CTRL.cls

```diff
@@ -84,56 +101,90 @@ public with sharing class CON_DeleteContactOverride_CTRL {
         set;
     }
 
+    /**  @description Component-friendly NPSP namespace */
+    public String namespace {
+        get {
+            if (namespace == null) {
+                namespace = UTIL_Namespace.getComponentNamespace();
+            }
+            return namespace;
+        }
+        set;
+    }
+
+    /**
+     * @description True if the current user has minimum access for Contact deletion with this page.
+     */
+    public Boolean hasMinimumAccess {get;set;}
+
     /*******************************************************************************************************
-    * @description Standard controller constructor. Queries for contact, verifies whether the contact or its
-    * account should be deleted.
+    * @description Standard controller constructor.
     */
     public CON_DeleteContactOverride_CTRL(ApexPages.StandardController stdController) {
+        contactId = stdController.getId();
         setNavigationParams(ApexPages.currentPage().getParameters());
+    }
+
+    /*******************************************************************************************************
+    * @description Required processing to check access and establish context. Queries for contact, verifies
+    * whether the contact or its account should be deleted.
+    * @return Boolean
+    */
+    public void preProcessDelete() {
+        hasMinimumAccess = contactOverrideSelector.canDeleteContactRecord(contactId);
+        if (!hasMinimumAccess) {
+            return;
+        }
+
+        contactRecord = contactOverrideSelector.queryContact(contactId);
+
+        isHouseholdAccount = false;
+        isOnlyContactInHousehold = false;
+        hasAccountDeleteAccess = contactOverrideSelector.canDeleteAccountRecord(contactRecord.AccountId);
```

Comment:
Running canDeleteContactRecord and then this method canDeleteAccountRecord queries the same table twice in the same method.  Could these queries be combined?  Also, see the other question about querying the UserRecordAccess table vs. checking SObject delete access when the ctrl is with sharing.